middleware/proxy/proxy.go

```diff
@@ -123,11 +123,13 @@ func Do(c *fiber.Ctx, addr string) error {
 	res := c.Response()
 	originalURL := utils.CopyString(c.OriginalURL())
 	defer req.SetRequestURI(originalURL)
-	req.SetRequestURI(addr)
+
+	copiedURL := utils.CopyString(addr)
+	req.SetRequestURI(copiedURL)
 	// NOTE: if req.isTLS is true, SetRequestURI keeps the scheme as https.
 	// issue reference:
 	// https://github.com/gofiber/fiber/issues/1762
-	if scheme := getScheme(utils.UnsafeBytes(addr)); len(scheme) > 0 {
+	if scheme := getScheme(utils.UnsafeBytes(copiedURL)); len(scheme) > 0 {
```

Comment:
```suggestion
	if scheme := getScheme(utils.UnsafeBytes(addr)); len(scheme) > 0 {
```
and this back to before

Reply:
This modification still has the original problem, and I haven't explored the cause. The `addr` will change after running `req.SetRequestURI` regardless of the data passed in.

I'll probably explore the reasons over the weekend.

Reply:
i was wrong -> your code is okay